app/src/org/commcare/dalvik/application/CommCareApp.java

```diff
@@ -131,8 +132,29 @@ public void setupSandbox(boolean createFilePaths) {
         }
     }
     
+    public void updateAppRecord() {
+        //uniqueId and displayName will be missing, so pull them from the app's profile file
+        record.setUniqueId(getUniqueId());
+        record.setDisplayName(getDisplayName());
+        //Similarly, this field may be incorrect
+        record.setResourcesStatus(areResourcesValidated());
+        //set this to false so we don't try to update this app record every time we seat it
+        record.setConvertedFromOldFormat(false);
+        //commit changes
+        CommCareApplication._().getGlobalStorage(ApplicationRecord.class).write(record);
+    }
     
     public boolean initializeApplication() {
+        boolean isOk = initializeApplicationHelper();

```

Comment:
isOk is kind of a generic description here. I might call this something less ambiguous like "isAppReady"